### PR TITLE
Add local display driver IL0323 for E-ink displays

### PR DESCRIPTION
**Pull request by LOWPROKB**
These files were taken verbatim from https://github.com/petejohanson/zephyr/commit/6a32cb14cc83241cea1bedb4098bc07b6a7b5fae

This commit adds driver files to the local zmk app directory in lieu of going through the process of an Upstream PR. It **will** get upstream'ed, but I do not have the time currently to do so.

First PR, so hopefully got the clang/prettier stuff working correctly.

### CODE REVIEW

File: app/drivers/display/Kconfig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2020 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+source "Kconfig.il0323"
```
Comment (petejohanson) [inline app/drivers/display/Kconfig:4]: This should be `rsource` to source a file relative to the current file.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Couple minor comments.

File: app/drivers/display/Kconfig (hunk 1 above)
Comment (LOWPROKB) [reply app/drivers/display/Kconfig:4]: ahhh... I was wondering what was going on there. I was dumbfounded as to what was going on and why I had to put the full local path to the file lol. Will fix this now.

File: app/drivers/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Copyright (c) 2020 The ZMK Contributors
 # SPDX-License-Identifier: MIT
 
+rsource "battery/Kconfig"
```
[the remark this answers is not in the document]
Comment (LOWPROKB) [reply app/drivers/CMakeLists.txt]: this has been changed

File: app/drivers/display/Kconfig (hunk 1 above)
Comment (LOWPROKB) [reply app/drivers/display/Kconfig]: this has been changed

File: app/drivers/display/il0323.c [hunk 3]
```diff
@@ -0,0 +1,467 @@
+/*
+ * Copyright (c) 2020 PHYTEC Messtechnik GmbH
```
Comment (petejohanson) [inline app/drivers/display/il0323.c:2]: This should now include me in the list, e.g.:

```suggestion
 * Copyright (c) 2020 PHYTEC Messtechnik GmbH, Peter Johanson
```

File: app/drivers/display/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/drivers/display/CMakeLists.txt:1]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

File: app/drivers/display/Kconfig [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/drivers/display/Kconfig:1]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

**Review by petejohanson — CHANGES_REQUESTED** [review]
Just a couple minor items left.

File: app/drivers/display/CMakeLists.txt (hunk 4 above)
Comment (LOWPROKB) [reply app/drivers/display/CMakeLists.txt]: done

File: app/drivers/display/Kconfig (hunk 5 above)
Comment (LOWPROKB) [reply app/drivers/display/Kconfig]: done

File: app/drivers/display/il0323.c (hunk 3 above)
Comment (LOWPROKB) [reply app/drivers/display/il0323.c]: done

**Review by petejohanson — APPROVED** [review]
Thanks! Tested here locally w/ an nRF52840 MDK dev kit along w/ a waveshare module on a breadboard.

**Comment by petejohanson** [comment]
Merged this manually after doing some clang-format clean up.

Thanks @LOWPROKB!

**Pull request closed without merging**